subsys/net/lib/azure_fota/azure_fota.c

```diff
@@ -0,0 +1,546 @@
+
+#define STATUS_CURRENT_STR "\"currentFwVersion\":\"%s\""
+#define STATUS_PENDING_STR "\"pendingFwVersion\":\"%s\""
+#define STATUS_UPDATE_STR "\"fwUpdateStatus\":\"%s\""
+#define STATUS_UPDATE_SUB_STR "\"fwUpdateSubstatus\":\"%s\""
+
+#define STATUS_REPORT					\
+	"{"						\
+		"\"firmware\":{"			\
+			STATUS_UPDATE_STR ","		\
+			STATUS_CURRENT_STR ","		\
+			STATUS_PENDING_STR		\
+		"}"					\
+	"}"
+
+/* Enum to keep the fota status */
+enum fota_status {
+	STATUS_CURRENT = 0,
+	STATUS_DOWNLOADING,
+	STATUS_APPLYING,
+	STATUS_REBOOTING,
+	STATUS_ERROR,
+	STATUS_ROLLEDBACK,
+
+	STATUS_COUNT,
+};
+
+struct fota_object {
+	char host[CONFIG_AZURE_FOTA_HOSTNAME_MAX_LEN];
+	char path[CONFIG_AZURE_FOTA_FILE_PATH_MAX_LEN];
+	char version[CONFIG_AZURE_FOTA_VERSION_MAX_LEN];
+	size_t fragment_size;
+};
+
+/* Static variables */
+static azure_fota_callback_t callback;
+static enum fota_status fota_state;
+static struct fota_object fota_object;
+static char *status_str[STATUS_COUNT] = {
+	[STATUS_CURRENT]	= "current",
+	[STATUS_DOWNLOADING]	= "downloading",
+	[STATUS_APPLYING]	= "applying",
+	[STATUS_REBOOTING]	= "rebooting",
+	[STATUS_ERROR]		= "error",
+	[STATUS_ROLLEDBACK]	= "rolledback",
+};
+
+#if IS_ENABLED(CONFIG_AZURE_FOTA_APP_VERSION_AUTO)
+static char current_version[CONFIG_AZURE_FOTA_VERSION_MAX_LEN] =
+	STRINGIFY(APP_VERSION);
+#else
+BUILD_ASSERT(sizeof(CONFIG_AZURE_FOTA_APP_VERSION) > 1, "No app version given");
+static char current_version[CONFIG_AZURE_FOTA_VERSION_MAX_LEN] =
+	CONFIG_AZURE_FOTA_APP_VERSION;
+#endif
+
+#if IS_ENABLED(AZURE_FOTA_TLS) && !defined(CONFIG_AZURE_FOTA_SEC_TAG)
+BUILD_ASSERT(0, "CONFIG_AZURE_FOTA_SEC_TAG must be defined");
+#endif
+
+static char *create_report(void)
+{
+	size_t buf_size = sizeof(STATUS_REPORT) +
+			2 * CONFIG_AZURE_FOTA_VERSION_MAX_LEN + 15;
+	ssize_t len;
+	char *buf = k_malloc(buf_size);
+
+	if (buf == NULL) {
+		LOG_WRN("Failed to allocate memory for FOTA report");
+		return NULL;
+	}
+
+	len = snprintk(buf, buf_size, STATUS_REPORT, status_str[fota_state],
+		       current_version, fota_object.version);
+	if ((len == 0) || (len >= buf_size)) {
+		LOG_WRN("Failed to create FOTA report");
+		k_free(buf);
+
+		return NULL;
+	}
+
+	LOG_DBG("Created FOTA report: %s", log_strdup(buf));
+
+	return buf;
+}
+
+static void callback_with_report(struct azure_fota_event *evt)
+{
+	evt->report = create_report();
+	if (evt->report == NULL) {
+		LOG_WRN("FOTA report was not created");
+	}
+
+	callback(evt);
+
+	if (evt->report) {
+		k_free(evt->report);
+	}
+}
+
+static char *get_current_version(void)
+{
+	return current_version;
+}
+
+static bool is_update(void)
+{
+	return strncmp(get_current_version(), fota_object.version,
+		       strlen(get_current_version())) != 0;
+}
+
+
+static void fota_dl_handler(const struct fota_download_evt *fota_evt)
+{
+	struct azure_fota_event evt;
+
+	switch (fota_evt->id) {
+	case FOTA_DOWNLOAD_EVT_FINISHED:
+		LOG_INF("FOTA download completed evt received");
```

Comment:
Could write without abbreviations here (evt). Alternatively, log the event names as they get propagated.

Reply:
I reduced the log level to debug, and left the content as is. I think it's okay to keep debug level stuff abbreviated